src/components/SerialConsole/SerialConsole.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import XTerm from './XTerm';
+
+const defaultMessages = {
+  textDisconnect: 'Disconnect',
+  textDisconnected:
+    'Disconnected from serial console. Click the Reconnect button.',
+  textLoading: 'Loading ...',
+  textReconnect: 'Reconnect'
+};
+
+class SerialConsole extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focusTerminal = this.focusTerminal.bind(this);
+    this.onResetClick = this.onResetClick.bind(this);
+    this.onDisconnectClick = this.onDisconnectClick.bind(this);
+
+    this.msg = this.msg.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.onConnect();
+  }
+
+  componentWillUnmount() {
+    this.props.onDisconnect();
+  }
+
+  msg(key) {
+    return this.props.messages[key] || defaultMessages[key];
+  }
+
+  onResetClick(event) {
+    if (event.button !== 0) return;
+
+    this.props.onDisconnect();
+    this.props.onConnect();
+    this.focusTerminal();
+  }
+
+  onDisconnectClick(event) {
+    if (event.button !== 0) return;
+
+    this.props.onDisconnect();
+    this.focusTerminal();
+  }
+
+  /**
+   * Backend sent data.
+   */
+  onDataReceived(data) {
+    if (this.refs.terminal) {
+      this.refs.terminal.onDataReceived(data);
+    }
+  }
+
+  /**
+   * Backend closed connection.
+   */
+  onConnectionClosed(reason) {
+    if (this.refs.terminal) {
+      this.refs.terminal.onConnectionClosed(reason);
+    }
+  }
+
+  focusTerminal() {
+    this.refs.resetButton.blur();
+    this.refs.disconnectButton.blur();
+
+    if (this.refs.terminal) this.refs.terminal.focus();
+  }
+
+  render() {
+    const { id, status, topClassName } = this.props;
+    const idPrefix = `${id || 'id'}-serialconsole`;
+
+    let terminal;
+    let disconnectDisabled = 'disabled'; // CSS class name
+    switch (status) {
+      case 'connected':
```

Comment:
maybe break out the statuses into a `constants.js`?

Reply:
done